### PR TITLE
php7 typehints for Elastica\Suggest namespace

### PR DESCRIPTION
**Pull request by osavchenko**
See #1548

### CODE REVIEW

File: lib/Elastica/Suggest/AbstractSuggest.php [hunk 1]
```diff
@@ -79,7 +79,7 @@ public function setRegexOptions(array $value)
      *
      * @return $this
      */
-    public function setField($field)
+    public function setField($field): AbstractSuggest
```
Comment (massimilianobraglia) [inline lib/Elastica/Suggest/AbstractSuggest.php]: missing `string` typehint

File: lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php [hunk 2]
```diff
@@ -39,7 +39,7 @@ public function setField($field)
      *
      * @return $this
      */
-    public function setSize($size)
+    public function setSize(int $size)
```
Comment (p365labs) [inline lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: default to ```5```

File: lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php [hunk 3]
```diff
@@ -49,7 +49,7 @@ public function setSize($size)
      *
      * @return $this
      */
-    public function setSuggestMode($mode)
+    public function setSuggestMode(string $mode)
```
Comment (p365labs) [inline lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: default to ```missing```

File: lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php [hunk 4]
```diff
@@ -59,7 +59,7 @@ public function setSuggestMode($mode)
      *
      * @return $this
      */
-    public function setMaxEdits($max)
+    public function setMaxEdits(int $max)
```
Comment (p365labs) [inline lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: Defaults to ```2```

File: lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php [hunk 5]
```diff
@@ -69,7 +69,7 @@ public function setMaxEdits($max)
      *
      * @return $this
      */
-    public function setPrefixLength($length)
+    public function setPrefixLength(int $length)
```
Comment (p365labs) [inline lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: Defaults to ```1```

File: lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php [hunk 6]
```diff
@@ -79,7 +79,7 @@ public function setPrefixLength($length)
      *
      * @return $this
      */
-    public function setMinWordLength($min)
+    public function setMinWordLength(int $min)
```
Comment (p365labs) [inline lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: min_word_length defaults value to ```4```

File: lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php [hunk 7]
```diff
@@ -89,7 +89,7 @@ public function setMinWordLength($min)
      *
      * @return $this
      */
-    public function setMaxInspections($max)
+    public function setMaxInspections(int $max)
```
Comment (p365labs) [inline lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: - max_inspections
- min_doc_freq
- max_term_freq
have their own default value

File: lib/Elastica/Suggest/Phrase.php [hunk 8]
```diff
@@ -39,7 +39,7 @@ public function setGramSize($size)
      *
      * @return $this
      */
-    public function setRealWordErrorLikelihood($likelihood)
+    public function setRealWordErrorLikelihood(float $likelihood): Phrase
```
Comment (p365labs) [inline lib/Elastica/Suggest/Phrase.php]: default value to ```0.95```

File: lib/Elastica/Suggest/Phrase.php [hunk 9]
```diff
@@ -52,7 +52,7 @@ public function setRealWordErrorLikelihood($likelihood)
      *
      * @return $this
      */
-    public function setConfidence($confidence)
+    public function setConfidence(float $confidence): Phrase
```
Comment (p365labs) [inline lib/Elastica/Suggest/Phrase.php]: The default is ```1.0```

File: lib/Elastica/Suggest/Phrase.php [hunk 10]
```diff
@@ -64,7 +64,7 @@ public function setConfidence($confidence)
      *
      * @return $this
      */
-    public function setMaxErrors($max)
+    public function setMaxErrors(float $max): Phrase
```
Comment (p365labs) [inline lib/Elastica/Suggest/Phrase.php]: The default is set to ```1.0```

File: lib/Elastica/Suggest/Term.php [hunk 11]
```diff
@@ -64,9 +64,9 @@ public function setLowercaseTerms($lowercase = true)
      *
      * @return $this
      */
-    public function setMaxEdits($max)
+    public function setMaxEdits(int $max): Term
```
Comment (p365labs) [inline lib/Elastica/Suggest/Term.php]: Defaults to ```2```

File: lib/Elastica/Suggest/Term.php [hunk 12]
```diff
@@ -122,7 +122,7 @@ public function setMinDocFrequency($min)
      *
      * @return $this
      */
-    public function setMaxTermFrequency($max)
+    public function setMaxTermFrequency(float $max): Term
```
Comment (p365labs) [inline lib/Elastica/Suggest/Term.php]: Defaults to ```0.01f```

File: lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php (hunk 7 above)
Comment (osavchenko) [reply lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: didn't get this point. we should make it like `setMaxInspections(int $max = null)` and check field and depends on field set default value?

Comment (massimilianobraglia) [reply lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: https://github.com/ruflin/Elastica/pull/1579#discussion_r243801054
@osavchenko checkout this discussion about the default values :)

Comment (osavchenko) [reply lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: So, I need to define all default values as constants like `const DEFAULT_MAX_INSPECTIONS = some_value`?

Comment (p365labs) [reply lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: yep that's the idea

Comment (massimilianobraglia) [reply lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php]: Exactly. This for every value required by @p365labs

**Comment by osavchenko** [comment]
@p365labs could you please review my PR?

**Pull request merged**